src/networking.c

```diff
@@ -1552,12 +1552,17 @@ void unlinkClient(client *c) {
          * in which case it needs to be cleaned from that list */
         if (c->flag.replica && c->repl_state == REPLICA_STATE_WAIT_BGSAVE_END && server.rdb_pipe_conns) {
             int i;
+            int still_alive = 0;
             for (i = 0; i < server.rdb_pipe_numconns; i++) {
                 if (server.rdb_pipe_conns[i] == c->conn) {
                     rdbPipeWriteHandlerConnRemoved(c->conn);
                     server.rdb_pipe_conns[i] = NULL;
```

Comment:
I wonder if we could also decrement the server.rdb_pipe_numconns here and swap the entry with the last one?
not that important but will reduce the need to iterate over null connections later on...

Reply:
What would be the benefit of incrementally cleaning up rdb_pipe_numconns ?

Reply:
yes, the decrement thing should work i guess, but i may not want to modify it, fear of that i will cause some troubles LOL.

Reply:
IMO - For such a small array, the overhead of managing swapping/shifting logic outweighs the benefit. Simplicity wins here.

Reply:
Fine, there goes away my wish to connect 1M replicas... :)